pkg/test/case.go

```diff
@@ -69,13 +70,26 @@ func (t *Case) DeleteNamespace(cl client.Client, ns *namespace) error {
 		defer cancel()
 	}
 
-	return cl.Delete(ctx, &corev1.Namespace{
+	nsObj := &corev1.Namespace{
 		ObjectMeta: metav1.ObjectMeta{
 			Name: ns.Name,
 		},
 		TypeMeta: metav1.TypeMeta{
 			Kind: "Namespace",
 		},
+	}
+
+	if err := cl.Delete(ctx, nsObj); err != nil {
+		return err
+	}
+
+	return wait.PollImmediateUntilWithContext(ctx, 100*time.Millisecond, func(ctx context.Context) (done bool, err error) {
+		actual := &corev1.Namespace{}
+		err = cl.Get(ctx, client.ObjectKey{Name: ns.Name}, actual)
+		if err == nil || !k8serrors.IsNotFound(err) {
+			return false, err
+		}
+		return true, nil
```

Comment:
```suggestion
		if k8serrors.IsNotFound(err) {
			return true, nil
		}
		return false, err
```

I find this a bit easier to read.

Reply:
👍